pkg/apis/logging/v1/output_types.go

```diff
@@ -0,0 +1,109 @@
+package v1
+
+import (
+	"github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1/outputs"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Output defines a destination for log messages.
+type Output struct {
+	// Name used to refer to the output from a `pipeline`.
+	//
+	// +required
+	Name string `json:"name"`
+
+	// OutputType is inlined with a required `type` and optional extra configuration.
+	OutputType `json:",inline"`
+
+	// Destination of log messages, an absolute URL (must have a scheme).
+	//
+	// Valid URL schemes depend on `type`.  Special schemes 'tcp', 'udp' and
+	// 'tls' are used for output types that don't define their own URL scheme.
+	// Example:
+	//
+	//     { type: syslog, destination: tls://syslog.example.com:1234 }
+	//
+	// TLS with server authentication is enabled by the URL scheme, for
+	// example 'tls' or 'https'.  See `secretRef` for TLS client authentication.
+	//
+	// +required
+	Destination string `json:"destination"`
```

Comment:
IMHO, destination is a configuration property of the output type. Although, it seems that we will have a connection string for all our output types, I can imagine more complicated destination properties, e.g. connecting through a sidecar to the target destination where your connection string could be a ref to a config map. Furthermore, it makes the YAML output more coherent, e.g.:
```
ouput:
   name: "my-target-syslog"
   syslog:
      destination: "tls://syslog.example.com:1234"
      facility: "whatever"
```

Reply:
disagree here.  the idea is top level fields are generic and should apply to any output followed by type specific fields.  In this case 'destination' is applicable to every output

Reply:
Strongly disagree - +1 Jeff. The goal is to avoid the anti-pattern adopted by most logging solutions where you can't do anything without learning type-specific details.  E.g.
```
output:
  type: kafka
  url: tls://foobar.com/my-topic
  secretRef: {...}
output:
  type: fluent
  url:tls://xyz.com:1234
  secretRef: {...} 
```  
You should not be required to learn that kafka expects TLS secrets as env. vars. naming PEM format cert files, but fluent expects blah blah blah. Things that are effectively the same for all output types should be configured the same in our API. Likewise for reconnect configuration when we add it - are reconnect intervals seconds or milliseconds or blah blah. It's all the same and we can provide common config for it.